### ISSUE TITLE
Don't warn about SC2046 (word splitting) for pseudo inline comments

### ISSUE THREAD

**Issue by Herst** [koalaman/shellcheck#729]
By using line comments inside command substitutions (described e.g. on http://stackoverflow.com/a/12797512/2261442) it's possible to have some sort of inline comments in shell scripting as well. This can then be used to e.g. comment out parameters or such:

```sh
foobar \
  --param1 \
  `# --param2 to be enabled once it's ready` \
  --param3
```

Shellcheck warns about possible word splitting and such because it's not inside double quotes, but this is exactly not what you would want because it would then create a parameter for the program which is empty but can still have an effect. I don't see how not showing the warning for empty command substitutions could hide any problems.

**Comment by Herst** [comment]
Same goes for the other method mentioned in the comments of the Stack Overflow post:
```sh
foobar \
  --param1 \
  ${IFS# --param2 to be enabled once it's ready} \
  --param3
```

**Comment by koalaman** [comment]
Indeed. 5a46eeb fixes the relatively common `` `#comment` `` case.

I haven't been able to find any real uses of `${IFS# ` comments, so those are left as-is.
